server/middleware/versions.coffee

```diff
@@ -89,7 +94,11 @@ module.exports =
 
     doc.set('parent', latest._id)
 
-    doc = yield doc.save()
+    try
+      doc = yield doc.save()
+    catch e
+      yield latest.update({$set: _.pick(latest.toObject(), 'version', 'index', 'slug')})
+      throw e
```

Comment:
The whole process was a bit confusing to figure out why we were re-assigning the version information to `latest` when we never make any changes to `latest` before this AFAICT.

I'm guessing that some request in here implicitly modifies `latest`'s `version.isLatestMajor`, but I don't really know.

Adding a couple explanatory comments about why/how we're changing the version info in various places would make this endpoint a bit easier to understand.